workers/generic-worker/payload_test.go

```diff
@@ -167,8 +168,32 @@ func TestArtifactExpiresBeforeDeadline(t *testing.T) {
 	ensureMalformedPayload(t, task)
 }
 
+// If a task has a higher `maxRunTime` than the `maxTaskRunTime` set in the worker config we should get a Malformed Payload
+func TestMaxTaskRunTime(t *testing.T) {
+	setup(t)
+	now := NowMillis(t)
+	task := taskWithPayload(`{
+  "env": {
+    "XPI_NAME": "dist/example_add-on-0.0.1.zip"
+  },
+  "maxRunTime": 86401,
+  "command": [` + rawHelloGoodbye() + `],
+  "artifacts": [
+    {
+      "type": "file",
+      "path": "public/some/artifact",
+      "expires": "` + tcclient.Time(now.Add(time.Minute*20)).String() + `"
+    }
+  ]
+}`)
+	task.Definition.Deadline = tcclient.Time(now.Add(time.Minute * 10))
+	task.Definition.Expires = tcclient.Time(now.Add(time.Minute * 20))
+	ensureMalformedPayload(t, task)
+}
```

Comment:
it would be very easy for this task to resolve as malformed payload because it contains a raw string which is quite complicated which is meant to be json. If there was a missing quotation mark etc, it would resolve as malformed payload. There are other tests that don't use raw strings, and create a very simple task. Would it be possible to carbon cult one of those instead? Another thing that would be useful here is checking that the error message contained the expected text about the maxRunTime being too high. There are other tests that check the log messages - can you use one of those as a basis? This way, we can be sure the malformed payload really is due to the maxRunTime and not one of the other causes of a malformed payload exception, such as invalid json, or if the syntax of artifacts changes, etc.

Reply:
Thanks for the feedback, that cleaned up nicely.

Reply:
Thanks for the feedback. I updated the test to declare the payload using `GenericWorkerPayload`struct instead of a raw string and refactored it so that it checks that the test fails due to a malformed payload exception caused by an out of bounds `MaxTaskRunTime`.